python/cudf/cudf/_lib/unary.pyx

```diff
@@ -93,14 +95,22 @@ def is_valid(Column input):
 
 def cast(Column input, object dtype=np.float64):
     cdef column_view c_input = input.view()
-    cdef type_id tid = (
-        <type_id> (
-            <underlying_type_t_type_id> (
-                np_to_cudf_types[np.dtype(dtype)]
+    cdef type_id tid
+    cdef data_type c_dtype
+
+    if is_decimal_dtype(dtype):
+        tid = libcudf_types.type_id.DECIMAL64
+        c_dtype = data_type(tid, -dtype.scale)
+    else:
+        tid = (
+            <type_id> (
+                <underlying_type_t_type_id> (
+                    np_to_cudf_types[np.dtype(dtype)]
+                )
             )
         )
```

Comment:
PR https://github.com/rapidsai/cudf/pull/7435/ introduces a Cython utility `dtype_to_data_type` that encapsulates  this logic. Can we add a TODO to use that when it becomes available?